rust/agama-server/src/manager/web.rs

```diff
@@ -226,36 +226,57 @@ async fn installer_status(
     Ok(Json(status))
 }
 
-/// Returns agama logs
-#[utoipa::path(get, path = "/api/manager/logs", responses(
-  (status = 200, description = "Download logs blob.")
-))]
+/// Creates router for handling /logs/* endpoints
+fn logs_router() -> Router<ManagerState<'static>> {
+    Router::new()
+        .route("/store", get(download_logs))
+        .route("/list", get(show_logs))
+}
 
-pub async fn download_logs() -> impl IntoResponse {
-    let path = generate_logs().await;
-    let Ok(path) = path else {
-        return (StatusCode::INTERNAL_SERVER_ERROR).into_response();
-    };
+#[utoipa::path(get, path = "/manager/logs/store", responses(
+    (status = 200, description = "Compressed Agama logs", content_type="application/octet-stream"),
+    (status = 500, description = "Cannot collect the logs"),
+    (status = 507, description = "Server is probably out of space"),
+))]
 
-    match ServeFile::new(path)
-        .try_call(Request::new(axum::body::Body::empty()))
-        .await
-    {
-        Ok(res) => res.into_response(),
-        Err(_) => (StatusCode::INTERNAL_SERVER_ERROR).into_response(),
-    }
-}
+async fn download_logs() -> impl IntoResponse {
+    let mut headers = HeaderMap::new();
+    let err_response = (headers.clone(), Body::empty());
 
-async fn generate_logs() -> Result<String, Error> {
-    let random_name: String = Alphanumeric.sample_string(&mut rand::thread_rng(), 8);
-    let path = format!("/run/agama/logs_{random_name}");
+    match store_logs() {
+        Ok(path) => {
+            if let Ok(file) = tokio::fs::File::open(path.clone()).await {
+                let stream = ReaderStream::new(file);
+                let body = Body::from_stream(stream);
+                let _ = std::fs::remove_file(path.clone());
 
-    Command::new("agama")
-        .args(["logs", "store", "-d", path.as_str()])
-        .status()
-        .await
-        .map_err(|e| ServiceError::CannotGenerateLogs(e.to_string()))?;
+                // See RFC2046, RFC2616 and
+                // https://www.iana.org/assignments/media-types/media-types.xhtml
+                // or /etc/mime.types
```

Comment:
I do not now how "standard" is the mime.types (outside of Linux), but we can find it out later.